graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -460,187 +429,224 @@ def _split_classes_by_kind(self, class_name_to_definition):
-
-            if linked_class is None:
-                raise AssertionError(u'Property "{}" is declared with type Link but has no '
-                                     u'linked class: {}'.format(name, property_definition))
-
-            if linked_class not in self._vertex_class_names:
-                is_linked_class_abstract = class_name_to_definition[linked_class]['abstract']
-                all_subclasses_are_vertices = True
-                for subclass in self._subclass_sets[linked_class]:
-                    if subclass != linked_class and subclass not in self.vertex_class_names:
-                        all_subclasses_are_vertices = False
-                        break
-                if not (is_linked_class_abstract and all_subclasses_are_vertices):
-                    raise AssertionError(u'Property "{}" is declared as a Link to class {}, but '
-                                         u'that class is neither a vertex nor is it an '
-                                         u'abstract class whose subclasses are all vertices!'
-                                         .format(name, linked_class))
-
-            qualifier = linked_class
-        elif type_id in COLLECTION_PROPERTY_TYPES:
-            if linked_class is not None and linked_type is not None:
-                raise AssertionError(u'Property "{}" unexpectedly has both a linked class and '
-                                     u'a linked type: {}'.format(name, property_definition))
-            elif linked_type is not None and linked_class is None:
-                # No linked class, must be a linked native OrientDB type.
-                validate_supported_property_type_id(name + ' inner type', linked_type)
-
-                qualifier = linked_type
-            elif linked_class is not None and linked_type is None:
-                # No linked type, must be a linked non-graph user-defined type.
-                if linked_class not in self._non_graph_class_names:
-                    raise AssertionError(u'Property "{}" is declared as the inner type of '
-                                         u'an embedded collection, but is not a non-graph class: '
-                                         u'{}'.format(name, linked_class))
-
-                qualifier = linked_class
-            else:
-                raise AssertionError(u'Property "{}" is an embedded collection but has '
-                                     u'neither a linked class nor a linked type: '
-                                     u'{}'.format(name, property_definition))
+    def _set_up_non_graph_elements(self, class_name_to_definition):
+        """Set up non graph elements."""
+        kind = SchemaElement.ELEMENT_KIND_NON_GRAPH
+        for class_name in self.non_graph_class_names:
+            # Create non-graph SchemaElement.
+            inheritance_set = self._inheritance_sets[class_name]
+            elem, links = _get_schema_element_and_links(class_name, class_name_to_definition,
```

Comment:
Is there a good reason for abbreviating `element` to `elem`? If not, let's not abbreviate.